### ISSUE TITLE
K8s at 1.25.4

### ISSUE THREAD

**Issue by agregory999** [oracle-quickstart/oci-cloudnative#381]
Do we need to update the auto-scaler if we want to use 1.25? 

https://github.com/oracle-quickstart/oci-cloudnative/blob/master/deploy/complete/terraform/oke-autoscaler.tf

**Comment by junior** [comment]
Unfortunately, the current versioning of the autoscaler does not allow you to automatically get the latest. I've worked in a new script that do best effort, which means that will try to use the matching images (k8s == OKE autoscaler), if not, use the latest known version. This will prevent issues with new versions and the only caveat would be that will not support new features until the correct image version is used. There's no APIs to get OKE autoscaler image version. That's the current: 
```
{
    "Repository": "us-ashburn-1.ocir.io/oracle/oci-cluster-autoscaler",
    "Tags": [
        "1.17-2021.02",
        "1.17-2021.03",
        "1.18-2021.02",
        "1.18-2021.03",
        "1.18.3-4",
        "1.19-2021.03",
        "1.19-2021.08.03",
        "1.19.1-2",
        "1.19.1-4",
        "1.19.1-8",
        "1.20.0-2",
        "1.20.0-4",
        "1.20.0-6",
        "1.21.1-1",
        "1.21.1-3",
        "1.22.2-4",
        "1.23.0-4",
        "1.24.0-5",
        "1.25.0-6"
    ]
}
```

Note that the number after the dash is not predictable (The number before is not either, but can workaround), that's why need the script to match.

**Comment by streamnsight** [comment]
You can use the 1.25 image for a 1.25 or 1.26 cluster right now.
Note that the cluster-autoscaler OCI repo is being merged in the kubernetes autoscaler public repo and in the future (starting at 1.27) the autoscaler will be built from there.
Beware also that when that happens, the provider is going to be `oci` as the 2 providers `oci` and `oci-oke` are being merged in to 1

**Comment by junior** [comment]
Stack fixed, updated and tested up to OKE 1.29
